.github/workflows/ros-pkg-workflow.yml

```diff
@@ -0,0 +1,231 @@
+name: Build and pack Ros packages 
+on:
+  workflow_call:
+    inputs:
+      deploy:
+        required: false
+        type: string
+        default: 'false'
+      release:
+        required: false
+        type: string
+        default: 'false'
+
+    secrets:
+      auto_commit_user:
+        required: true
+      auto_commit_mail:
+        required: true
+      registry_user:
+        required: true
+      registry_password:
+        required: true
+      nexus_publisher_user:
+        required: true
+      nexus_publisher_password:
+        required: true
+      aws_sqs_rosdep_access_key:
+        required: true
+      aws_sqs_rosdep_secret_access_key:
+        required: true
+      github_token:
+        required: true
+      sonar_token:
+        required: true
+
+jobs:
+  Test-Ros-Component:
+    if: ${{ inputs.release == 'false' }}
+    runs-on: ubuntu-20.04
+    container:
+      image: registry.cloud.mov.ai/devops/ros-buildtools-melodic:main
+      options: --user root
+      credentials:
+        username: ${{secrets.registry_user}} 
+        password: ${{secrets.registry_password}}
+
+    steps:
+    - name: Checkout    
+      uses: actions/checkout@v2
+      with:
+        # Disabling shallow clone is recommended for improving relevancy of reporting
+        fetch-depth: 0
+        submodules: recursive
+
+    - name: Validate Repository Package.xml
+      run: |
+        expected_url="<url>$GITHUB_SERVER_URL/$GITHUB_REPOSITORY</url>"
+        VALIDATION_FILE="/tmp/verification-failed"
+        find -L ./ -type f -name 'package.xml' -exec sh -c '
+
+        search_result=$(cat $0 | grep "$1")
+
+        if [ -z "$search_result" ]:
+        then
+            echo "\e[31m$0"
+            touch $2
+        fi 
+
+        ' {} $expected_url $VALIDATION_FILE \;
+
+        if [ -f $VALIDATION_FILE ];
+        then    
+            echo "\e[31mPlease add \033[0;36m$expected_url \e[31mto all the package.xmls mentioned above!"
+            exit 1
+        fi 
+
+    - name: SonarQube Scan
+      uses: SonarSource/sonarqube-scan-action@v1.0.0
+      env:
+        SONAR_TOKEN: ${{ secrets.sonar_token }}
+        SONAR_HOST_URL: https://sonarqube.aws.cloud.mov.ai
+      with:
+        projectBaseDir: ./
+        args: >
+          -Dsonar.organization=MOV-AI
+          -Dsonar.projectKey=${{ github.event.repository.name }}
+          -Dsonar.verbose=true
+          -Dsonar.sources=.
+          -Dsonar.scm.provider=git
+          -Dsonar.qualitygate.wait=true
+          -Dsonar.qualitygate.timeout=300
+
+    - name: Link to SonarQube Dashboard
+      shell: bash
+      run: |
+        echo "Please check report here: https://sonarqube.aws.cloud.mov.ai/dashboard?id=${{ github.event.repository.name }}"
+      id: sonar_links
+
+    - name: Install Mobros
+      run: python3 -m pip install mobros==1.0.1.8 
```

Comment:
can we insert that inside the docker image instead ?

Reply:
In my opinion, would be easier to simply bump here as this is the one that has the main behavior should mainly be covered by mobros. With this we can simply bump the template with a new behavior without the need to rebuild the docker image. 
mobros install time = 5 secs,
docker image build time = 5 mins +  (plus promotion and so on)

Reply:
ok